script.js

```diff
@@ -1,12 +1,421 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+
+function getData(URL) {
+  fetch(URL)
+    .then((response) => response.json())
+    .then((json) => {
+      makeShowPage(json);
+      selectShow(json);
+      search();
+      result();
+      home();
+      updateResult();
+      selectEpisode(json);
+    });
+}
+
+function setup(URL) {
+  fetch(URL)
+    .then((response) => response.json())
+    .then((json) => {
+      makeEpisodePage(json);
+      selectEpisode(json);
+    });
 }
 
-function makePageForEpisodes(episodeList) {
+function makeEpisodePage(list) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  list.forEach((episode) => {
+    let eachDiv = document.createElement(`div`);
+    let containerDiv = rootElem.appendChild(eachDiv);
+    containerDiv.setAttribute(`id`, `${episode.id}`);
+    containerDiv.setAttribute(`class`, `container episode`);
+    containerDiv.setAttribute(`style`, `display: initial;`);
+
+    //take the name of each episode and put it inside containerDiv
+    let nameParagraph = document.createElement(`p`);
+    containerDiv.appendChild(nameParagraph);
+    nameParagraph.setAttribute(
+      `id`,
+      `nameS${codeCorrection(episode.season)}E${codeCorrection(episode.number)}`
+    );
+    nameParagraph.setAttribute(`class`, `name`);
+    nameParagraph.textContent = `${episode.name}`;
+
+    //take the episode code
+    let episodeCodeP = document.createElement(`p`);
+    containerDiv.appendChild(episodeCodeP);
+    episodeCodeP.setAttribute(
+      `id`,
+      `episodeCodeS${codeCorrection(episode.season)}E${codeCorrection(
+        episode.number
+      )}`
+    );
+    episodeCodeP.setAttribute(`class`, `episodeCode`);
+    episodeCodeP.textContent = `Episode Code: S${codeCorrection(
+      episode.season
+    )}E${codeCorrection(episode.number)}`;
+
+    //take the episode Images
+    let episodeImg = document.createElement(`img`);
+    let divImg = document.createElement(`div`);
+    divImg.appendChild(episodeImg);
+    containerDiv.appendChild(divImg);
+    divImg.setAttribute(`class`, `divImg`);
+    episodeImg.setAttribute(
+      `id`,
+      `episodeImgS${codeCorrection(episode.season)}E${codeCorrection(
+        episode.number
+      )}`
+    );
+    episodeImg.setAttribute(`class`, `episodeImg`);
+    episodeImg.src = episode.image.medium;
+    episodeImg.alt = `image`;
```

Comment:
the alt text should help people with visual difficulties understand what the image is. Perhaps this should be 'series poster'